library/general/disallowedtags/template.yaml

```diff
@@ -23,20 +23,29 @@ spec:
               description: Disallowed container image tags.
               items:
                 type: string
+            allowedImages:
+              type: array
+              description: Allowed container images.
+              items:
+                type: string
   targets:
     - target: admission.k8s.gatekeeper.sh
       rego: |
         package k8sdisallowedtags
 
         violation[{"msg": msg}] {
           container := input_containers[_]
+          images := [allowed | image = input.parameters.allowedImages[_]; allowed = (image == container.image) ]
+          not all(images)
```

Comment:
It looks like what you're trying to do is scroll through the list of exceptions, marking true if an exception is matched, false otherwise, then if there is at least one `true`, the exception is granted. Is that accurate?

If so, I think the correct word would be `not any(images)`

Also, nit on the variable name `images`, since it's not storing a list of images, maybe: `allowed_images_matches`?

Reply:
Then again `not any()` would fail in the case where there were no allowed images

Reply:
Maybe building the allowedImages check in a function would be easier? You could use the piecewise function behavior to emulate or-style logic.

Reply:
I updated tests with 2 containers checking:
- `exempt-image:disallowed-tag` and `another-image:accept-tag` should have no violation
- `exempt-image:disallowed-tag` and `another-image:disallowed-tag` should be violated

strangely they both passed the test under `not any(matched_allowed_images)` and  `not all(matched_allowed_images)`

Reply:
Sorry, was overthinking with `not any`, if there are no allowed images, this should return `true` always, which is what we want.

Reply:
The difference in behavior between `any` and `all` only shows up if there are multiple allowed images.

Reply:
Updated the tests with 2 allowed images, turns out `not all` is not a valid.